server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -312,10 +313,14 @@ public ClusterState execute(ClusterState currentState) {
                                                     IndexMetadata.Builder indexMdBuilder = IndexMetadata.builder(snapshotIndexMetadata)
                                                         .state(IndexMetadata.State.OPEN)
                                                         .index(renamedIndexName);
-                                                    indexMdBuilder.settings(Settings.builder()
-                                                                                .put(snapshotIndexMetadata.getSettings())
-                                                                                .put(IndexMetadata.SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
-                                                                                .put(IndexMetadata.SETTING_HISTORY_UUID, UUIDs.randomBase64UUID()));
+                                                    Builder indexSettingsBuilder = Settings.builder()
+                                                        .put(snapshotIndexMetadata.getSettings())
+                                                        .put(IndexMetadata.SETTING_INDEX_UUID, UUIDs.randomBase64UUID());
+                                                    if (snapshotIndexMetadata.getCreationVersion().onOrAfter(Version.V_5_1_0)
```

Comment:
5.2.0 (because we haven't backported this to 5.1)?

Reply:
We introduced the HISTORY UUID in 5.1